lkmpg.tex

```diff
@@ -927,8 +930,8 @@ \subsection{chardev.c}
 In the multiple-threaded environment, without any protection, concurrent access to the same memory may lead to the race condition, and will not preserve the performance.
 In the kernel module, this problem may happen due to multiple instances accessing the shared resources.
 Therefore, a solution is to enforce the exclusive access.
-We use atomic Compare-And-Swap (CAS), the single atomic operation, to determine whether the file is currently open by someone.
-CAS compares the contents of a memory loaction with the expected value and, only if they are the same, modifies the contents of that memory location to the desired value.
+We use atomic Compare-And-Swap (CAS) to determine whether the file is currently opened by someone.
+CAS compares the contents of a memory location with the expected value and, only if they are the same, modifies the contents of that memory location to the desired value.
```

Comment:
In TeX document, you should explain the use of atomic operations in the above LKM, at least for `CDEV_NOT_USED` and `CDEV_EXCLUSIVE_OPEN`.